### PR TITLE
read firmware verion for FLORA devices

### PR DESCRIPTION
**Pull request by d0m1n1qu3**
## Description:

the FLORA sensors send the firmware version number while they send the
battery info so we parse it and send it via mqtt (json) and also
present it in the webinterface

only in files for ESP32

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on core ESP8266 V.2.7.2
  - [x] The code change is tested and works on core ESP32 V.1.12.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by d0m1n1qu3** [comment]
here a screenshot :-)

image

**Comment by Staars** [comment]
There is nothing wrong with this approach and especially on the ESP32 we have some room left, but please ask yourself, if you really need a special feature. 
It is not so hard to read all kinds of info for every known device (I have 8 different types at home) but at some point in the future, we will have a memory problem.
I think the hardest challenge in the long run is to leave things out.
(Again: I have no problem with a merge)

**Comment by d0m1n1qu3** [comment]
i found it really helpful to know which firmware the device have because i had problems with some older firmwares. 
and i like the idea to use tasmota to find out the firmware version without install apps or other tools. 
but i will keep it in my mind. :-)

**Comment by arendst** [comment]
A better solution would be to drop static info from the gui and use mqtt only. This way you still have the info you need while the gui stays clean

**Comment by d0m1n1qu3** [comment]
but this info is only displayed if you send the command MI32Battery
question: is it a display problem because of full ui or a memory problem because of the "const char HTTP_FIRMWARE[] PROGMEM"?

**Comment by arendst** [comment]
On an esp32 it's a gui problem for now. I didn't see you did add a command too. In that case I'll merge.

**Comment by d0m1n1qu3** [comment]
i use the command for the battery call because you get the frimware version in the same data packet from the sensor

File: tasmota/xsns_62_MI_ESP32.ino [hunk 1]
```diff
@@ -1012,6 +1015,23 @@ bool MI32readBat(char *_buf){
...
+    char _firmware[5]; // FLORA send 5 byte for firmware version
+    strncpy(_firmware, _buf+2, 5);
+    AddLog_P2(LOG_LEVEL_DEBUG,PSTR("%s: Firmware: %s"),D_CMND_MI32,_firmware);
+
+    uint32_t _slot = MI32.state.sensor;
+    DEBUG_SENSOR_LOG(PSTR("MIBLE: Sensor slot: %u"), _slot);
+
+    memcpy(MIBLEsensors[_slot].firmware, _firmware, 5);
+    MIBLEsensors[_slot].firmware[5] = '\0';
+    return true;
```
Comment (arendst) [inline tasmota/xsns_62_MI_ESP32.ino:1030]: There is no firmware[5] on a defined char of [5].

**Comment by arendst** [comment]
Battery level is fine as dynamic. Firmware being statis should not be on the gui

**Comment by d0m1n1qu3** [comment]
but i and a mate from me like it on the gui :-) 
than i change it to mqtt only?

**Comment by arendst** [comment]
Yes mqtt only. Thx in advance.

**Pull request merged**